rpc/client/helpers.go

```diff
@@ -52,32 +53,24 @@ func WaitForHeight(c StatusClient, h int64, waiter Waiter) error {
 	return nil
 }
 
-// WaitForOneEvent subscribes to a websocket event for the given
-// event time and returns upon receiving it one time, or
-// when the timeout duration has expired.
-//
-// This handles subscribing and unsubscribing under the hood
-func WaitForOneEvent(c SubscriptionClient, evtTyp string, timeout time.Duration) (types.TMEventData, error) {
-	const subscriber = "helpers"
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
-
-	// register for the next event of this type
-	eventCh, err := c.Subscribe(ctx, subscriber, types.QueryForEvent(evtTyp).String())
-	if err != nil {
-		return nil, fmt.Errorf("failed to subscribe: %w", err)
-	}
-	// make sure to unregister after the test is over
-	defer func() {
-		if deferErr := c.UnsubscribeAll(ctx, subscriber); deferErr != nil {
-			panic(deferErr)
+// WaitForOneEvent waits for the first event matching the given query on c, or
+// until ctx ends. It reports an error if ctx ends before a matching event is
+// received.
+func WaitForOneEvent(ctx context.Context, c EventsClient, query string, evt types.TMEventData) error {
+	for {
+		rsp, err := c.Events(ctx, &coretypes.RequestEvents{
+			Filter:   &coretypes.EventFilter{Query: query},
+			MaxItems: 1,
+			WaitTime: 10 * time.Second, // duration doesn't matter, limited by ctx timeout
+		})
+		if err != nil {
+			return err
+		} else if len(rsp.Items) == 0 {
+			continue // continue polling until ctx expires
```

Comment:
Not sure if it would be better to wait between polling but I guess it's fine

Reply:
Actually `WaitForOneEvent` keep continue since err set to [nil](https://github.com/tendermint/tendermint/blob/fe180ddd7b0a96363cb0e2c8567a7a302cfb8ab7/rpc/core/events.go#L231) when timeout.

Reply:
Do we need further change? Since [rpc](https://github.com/tendermint/tendermint/blob/fe180ddd7b0a96363cb0e2c8567a7a302cfb8ab7/rpc/core/events.go#L229) doesn't report a timeout as a request failure and this function seems only use for test.

Reply:
No I think it's okay. As you said, this is just used for testing